scheduler/scheduler.go

```diff
@@ -139,11 +139,11 @@ func (fq *fifoQueue) close() {
 func (fq *fifoQueue) reconfigure(c Config) {
 	fq.mu.Lock()
 	defer fq.mu.Unlock()
-	fq.maxConcurrency = uint64(c.MaxConcurrency)
-	fq.maxQueueSize = uint64(c.MaxQueueSize)
+	fq.maxConcurrency = int64(c.MaxConcurrency)
+	fq.maxQueueSize = int64(c.MaxQueueSize)
 	fq.timeout = c.Timeout
 	fq.sem = semaphore.NewWeighted(int64(c.MaxConcurrency))
-	fq.counter = atomic.NewUint64(0)
+	fq.counter = &atomic.Int64{}
```

Comment:
Lets use one of `&atomic.Int64{}` or `new(atomic.Int64)` consistently

Reply:
Patched to `new(atomic.Int64)` in the next patch.